kernel/src/process/signal/sig_stack.rs

```diff
@@ -1,99 +1,114 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use crate::prelude::*;
+use crate::{prelude::*, process::signal::c_types::stack_t};
 
-/// User-provided signal stack. `SigStack` is per-thread, and each thread can have
-/// at most one `SigStack`. If one signal handler specifying the `SA_ONSTACK` flag,
-/// the handler should be executed on the `SigStack`, instead of on the default stack.
+/// User-provided signal stack.
 ///
-/// SigStack can be registered and unregistered by syscall `sigaltstack`.
-#[derive(Debug, Clone)]
+/// Signal stack is per-thread, and each thread can have at most one signal stack.
+/// If one signal handler specifying the `SA_ONSTACK` flag,
+/// the handler should be executed on the signal stack, instead of on the default stack.
+///
+/// Signal stack can be registered and unregistered by syscall `sigaltstack`.
+#[derive(Debug, Default)]
 pub struct SigStack {
     base: Vaddr,
     flags: SigStackFlags,
     size: usize,
-    /// The number of handlers that are currently using the stack
-    handler_counter: usize,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum SigStackStatus {
+    /// The stack is enabled but currently inactive.
+    Inactive,
+    /// The stack is currently active.
+    Active,
+    /// The stack is disabled.
+    Disable,
+}
+
+impl From<SigStackStatus> for SigStackFlags {
```

Comment:
Is this `From` impl still useful now? This seems to be redundant given `impl From<SigStackStatus> for SigStackStatusFlags`.